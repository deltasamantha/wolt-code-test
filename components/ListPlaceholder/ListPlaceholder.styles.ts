import {StyleSheet} from "react-native";

import theme from "../../theme/theme";

export const styles = StyleSheet.create({
  rootContainer: {
    backgroundColor: theme.colors.colorWhite,
    minHeight: 200,
    alignItems: "center",
    justifyContent: "center",
  },
});
