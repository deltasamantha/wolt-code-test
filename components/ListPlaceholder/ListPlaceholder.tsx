import React from "react";
import {View} from "react-native";

import {styles} from "./ListPlaceholder.styles";
import locale from "../../config/Locale";
import AppText from "../AppText/AppText";

const ListPlaceholder = () => {
  return (
    <View style={styles.rootContainer}>
      <AppText size="m" color="light">
        {locale.slotListPlaceholder}
      </AppText>
    </View>
  );
};

export default ListPlaceholder;
