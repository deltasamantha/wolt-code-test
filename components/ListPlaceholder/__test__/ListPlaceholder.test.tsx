import {render, cleanup, screen} from "@testing-library/react-native";

import locale from "../../../config/Locale";
import ListPlaceholder from "../ListPlaceholder";

afterEach(cleanup);

describe("ListPlaceholder", () => {
  it("Should render ListItem properly", () => {
    const {toJSON} = render(<ListPlaceholder />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("ListItem should have placeholder text", () => {
    const placeholder = locale.slotListPlaceholder;
    render(<ListPlaceholder />);
    const placeholderText = screen.getByText(placeholder);
    expect(placeholderText.props.children).toBeDefined();
    expect(placeholderText.props.children).toEqual(placeholder);
  });
});
