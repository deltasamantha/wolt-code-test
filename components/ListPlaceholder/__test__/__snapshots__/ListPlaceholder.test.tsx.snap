// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPlaceholder Should render ListItem properly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "justifyContent": "center",
      "minHeight": 200,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#A1A2A4",
          "fontFamily": "Roboto_400Regular",
          "fontSize": 16,
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    Sorry we don't have any slots available!
  </Text>
</View>
`;
